bin/main

```diff
@@ -1,3 +1,118 @@
 #!/usr/bin/env ruby
 
-puts 'Hello world!'
+# welcome message
+def welcome_message
+  puts <<~HEREDOC
+    *****--welcome to Azeez and Helcio tic tac toe--*****
+         press any key to start the game and have fun!
+  HEREDOC
+end
+
+# player name
+def player_names
+  players = []
+  puts 'player one name?'
+  players[0] = gets.chomp
+
+  puts 'player two name?'
+  players[1] = gets.chomp
+
+  players
+end
+
+# def user_input()
+
+# end
+
+# border display
+def display_border(names)
+  p "#{names[0]} choose a valid number between 1 and 9!"
+
+  p ''
+
+  puts <<-HEREDOC
+    1 | 2 | 3
+  ---------------
+    4 | 5 | 6
+  ---------------
+    7 | 8 | 9
+  HEREDOC
+end
+
+def user_input
+  p 'choose a number between 1 and 9'
+  gets.chomp.to_i
+end
+
+def check_user_input(user_input)
+  if !user_input.between?(1, 9)
+    p 'please choose a valid number between 1 9 '
+  else
+    p 'you have printed a valid number'
+  end
+  user_input
+end
+
+# players symbols
+def players_symbols(players)
+  puts <<~HEREDOC
+    #{players[0]} symbol is X and #{players[1]} is O
+  HEREDOC
+end
+
+def check_empty_cell(row, column)
+  if play[row][column] != 'x' && play[row][column] != 'o'
+    play[row][column] = i.even? ? 'x' : 'o'
+  else
+    p 'This section is already filled'
+  end
+end
+
+welcome_message
+key = gets
+
+names = player_names if key
+
+players_symbols(names)
+
+display_border(names)
+
+check_user_input(user_input)
+play = [[' ', ' ', ' '], [' ', ' ', ' '], [' ', ' ', ' ']]
+gameover = false
+until gameover
+  puts '   1 2 3 '
+  puts '  -------'
+  puts "a |#{play[0][0]}|#{play[0][1]}|#{play[0][2]}|"
+  puts '  -------'
+  puts "b |#{play[1][0]}|#{play[1][1]}|#{play[1][2]}|"
+  puts '  -------'
+  puts "c |#{play[2][0]}|#{play[2][1]}|#{play[2][2]}|"
+  puts '  -------'
+  puts
+
+  input = gets.chomp
+  case input
+  when '1'
+    check_empty_cell(0, 0)
+  when '2'
+    check_empty_cell(0, 1)
+  when '3'
+    check_empty_cell(0, 2)
+  when '4'
+    check_empty_cell(1, 0)
+  when '5'
+    check_empty_cell(1, 1)
+  when '6'
+    check_empty_cell(1, 2)
+  when '7'
+    check_empty_cell(2, 0)
+  when '8'
+    check_empty_cell(2, 1)
+  when '9'
+    check_empty_cell(2, 2)
+  else
+    p 'Please enter a valid input between 1-9'
+  end
```

Comment:
- [ ] game logic is for the next step, please remove this.

Reply:
The last reviewer asked us to add that to the code. @spayco you guys are making this review complex